packages/core/src/modules/connections/handlers/ConnectionResponseHandler.ts

```diff
@@ -1,22 +1,44 @@
 import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { Handler, HandlerInboundMessage } from '../../../agent/Handler'
+import type { OutOfBandService } from '../../oob/OutOfBandService'
 import type { ConnectionService } from '../services/ConnectionService'
 
 import { createOutboundMessage } from '../../../agent/helpers'
+import { AriesFrameworkError } from '../../../error'
 import { ConnectionResponseMessage } from '../messages'
 
 export class ConnectionResponseHandler implements Handler {
   private connectionService: ConnectionService
+  private outOfBandService: OutOfBandService
+
   private agentConfig: AgentConfig
   public supportedMessages = [ConnectionResponseMessage]
 
-  public constructor(connectionService: ConnectionService, agentConfig: AgentConfig) {
+  public constructor(
+    connectionService: ConnectionService,
+    outOfBandService: OutOfBandService,
+    agentConfig: AgentConfig
+  ) {
     this.connectionService = connectionService
+    this.outOfBandService = outOfBandService
     this.agentConfig = agentConfig
   }
 
   public async handle(messageContext: HandlerInboundMessage<ConnectionResponseHandler>) {
-    const connection = await this.connectionService.processResponse(messageContext)
+    if (!messageContext.recipientVerkey || !messageContext.senderVerkey) {
+      throw new AriesFrameworkError('Unable to process connection request without senderVerkey or recipientVerkey')
+    }
+
+    const connectionRecord = await this.connectionService.findByVerkey(messageContext.recipientVerkey)
+    if (!connectionRecord) {
+      throw new AriesFrameworkError(`Connection for verkey ${messageContext.recipientVerkey} not found!`)
+    }
+
+    const outOfBandRecord =
```

Comment:
```suggestion
    const outOfBandRecord = connectionRecord.outOfBandId ? (await this.outOfBandService.findById(connectionRecord.outOfBandId)) : undefined
```

Reply:
This won't work because `findById` can return `null`. However, I hope I'll get rid of it when OOB replaces the old invitation.